workflows/data_pipelines/spectacle/spectacle_processor.py

```diff
@@ -0,0 +1,53 @@
+from dag_datalake_sirene.helpers import DataProcessor, Notification
+
+
+class SpectacleProcessor(DataProcessor):
+    def __init__(self):
+        from dag_datalake_sirene.workflows.data_pipelines.spectacle.spectacle_config import (
+            SPECTACLE_CONFIG,
+        )
+
+        super().__init__(SPECTACLE_CONFIG)
+
+    def preprocess_data(self):
+        import pandas as pd
+        from airflow.operators.python import get_current_context
+
+        df_spectacle = pd.read_csv(
+            self.config.file_download_destination, dtype=str, sep=";"
+        )
+
+        df_spectacle = (
+            df_spectacle.assign(
+                statut_du_recepisse=lambda x: x["Statut du récépissé"],
+                est_entrepreneur_spectacle=True,
+                siren=lambda x: x[
+                    "SIRET (personne morale) / SIREN (personne physique)"
+                ].str[:9],
+            )
+            .loc[
+                lambda x: x["siren"].notna() & x["siren"].str.isdigit(),
+                ["siren", "statut_du_recepisse", "est_entrepreneur_spectacle"],
+            ]
+            .groupby(["siren", "est_entrepreneur_spectacle"])
+            # If at least one of `statut` values is valid, then the value we keep is `valide
+            .agg(
+                statut_entrepreneur_spectacle=(
+                    "statut_du_recepisse",
+                    lambda x: "valide" if "Valide" in x.unique() else "invalide",
+                )
+            )
+            .reset_index()
+        )
+
+        df_spectacle[
+            ["siren", "statut_entrepreneur_spectacle", "est_entrepreneur_spectacle"]
+        ].to_csv(self.config.file_output, index=False)
+
+        ti = get_current_context()["ti"]
+        ti.xcom_push(
+            key=Notification.notification_xcom_key,
+            value=f"{str(df_spectacle.siren.nunique())} unités légales représentées.",
+        )
+
+        del df_spectacle
```

Comment:
In other data sources, I added a method like this :
```
def send_file_to_minio(self):
        super().send_file_to_minio()
        ti = get_current_context()["ti"]
        nb_siret = ti.xcom_pull(key="nb_siret_spectacle", task_ids="preprocess_spectacle")
        message = f"{nb_siret} établissements"
        ti.xcom_push(key=Notification.notification_xcom_key, value=message)
```
It sends a notif to Tchap with number of siren/siret processed.

Reply:
Indeed, I feel like it is not really this task job to manage this.
For another time, but we could also push such metric to our own dashboarding system or Kibana to monitor it better than in Tchap. What do you think?

Reply:
Yes, it's a good idea. The purpose for Tchap is not necessarily detail monitoring but rather raising alerts if that makes sense.

Reply:
Sowhat do we do about this : `ti.xcom_push(key=Notification.notification_xcom_key, value=message)`

Reply:
Can we leave it to `preprocess_data` to push this message to Tchap for now?
It is closer to where the metric has been created and I feel like using xcoms for this adds some unecessary complexity. Especially since we will have an error when `send_file_to_minio` fails anyway.
And I can create a ticket so we later work on storing such metrics in another place than Tchap?
What do you think?

Reply:
I agree, I don't have an issue with placement. My issue is about the text shown in Tchap. What's it going to look like ?

Reply:
True good point!
What do you think if we do?
```
    @staticmethod
    def _push_unique_count(column, xcom_key, description=None):
        """
        Counts unique values in the specified column and pushes the count to XCom.    
        If description is None, defaults to the column name.         

        Args:
            column (pd.Series): The Dataframe column to count unique values from.
            xcom_key (str): The key to use for pushing the count to XCom.
            description (str): The description to append next to the count.
        """
        unique_count = column.nunique()
        unique_count_str = f"{unique_count} {' ' + description if description else column.name}"
        ti = get_current_context()["ti"]
        ti.xcom_push(key=xcom_key, value=unique_count_str)
        logging.info(f"Processed {unique_count_str} unique values for {xcom_key}.")
```

Reply:
lgtm !